src/modules/trade/actions/place-trade.js

```diff
@@ -17,12 +18,16 @@ export const placeTrade = (marketId, outcomeId, tradeInProgress, doNotCreateOrde
     return dispatch(clearTradeInProgress(marketId))
   }
   const bnAllowance = createBigNumber(loginAccount.allowance)
-  // try and make sure that we actually have an updated allowance.
-  if (bnAllowance.lte(0)) dispatch(checkAccountAllowance())
+  const sharesDepleted = createBigNumber(tradeInProgress.sharesDepleted, 10)
+  const otherSharesDepleted = createBigNumber(tradeInProgress.otherSharesDepleted, 10)
+  const sharesProvided = sharesDepleted.eq(ZERO) ? otherSharesDepleted.toFixed() : sharesDepleted.toFixed()
+  // make sure that we actually have an updated allowance.
+  if (bnAllowance.lte(ZERO)) dispatch(checkAccountAllowance())
```

Comment:
Is this async? If so, should we put the rest of this in a callback?

Reply:
separate issue, planning to look at that today, it's the approval thing. That will probably go away or change slightly. That dispatch is intended to make sure we actually pulled the latest allowance but you are correct it shouldn't be async and that's probably what will change. feel free to ignore here though as it'll be in it's own PR